### DISCUSSION TITLE
Gatsby build failing with mozjpeg

### DISCUSSION THREAD

**Discussion by mnikolaus** [gatsbyjs/gatsby#28200] [Help]
From yesterday I am unable to deploy my Gatsby site to our server, build is complaining about mozjpeg library. Have anyone encountered this error before?

I tried:
- locking gatsby-cli
- adding `automake` package to dependency
- unfreezing yarn lock
- installing dependencies without lock file

But nothing seemed to help, the error is as follows:

```
error /node_modules/mozjpeg: Command failed.
Exit code: 1
Command: node lib/install.js
Arguments:
Directory: /node_modules/mozjpeg
Output:
⚠ spawn /node_modules/mozjpeg/vendor/cjpeg ENOENT
⚠ mozjpeg pre-build test failed
ℹ compiling from source
✖ Error: Command failed: /bin/sh -c ./configure --enable-static --disable-shared --disable-dependency-tracking --with-jpeg8 --prefix="/node_modules/mozjpeg/vendor" --bindir="/node_modules/mozjpeg/vendor" --libdir="/node_modules/mozjpeg/vendor"
```

**OS**
Ubuntu 16.04.6
**Node**
12.14
**Gatsby**
2.19.7

**Comment by ascorbic** [comment]
I'm not sure why `mozjpeg` isn't available as a pre-compiled module, but the issue does seem to be the compilation. Could you try installing it as a direct dependency?

**Comment by wardpeet** [comment]
@mnikolaus make sure to install build-essentials or something similar for the platform you're using.

**Comment by mnikolaus** [comment]
Installing `automake libtool dpkg pkgconfig nasm libpng` did solve the issue finally, thanks for you input.

**Comment by spectras** [comment]
Is there a way to skip building mozjpeg entirely?

This greatly increases the pipeline runtime, and it is not even used in the end. Not to mention the requirements of having all the build toolchain in the docker image.
